### PR TITLE
Add in/out message interceptors

### LINKED ISSUE #160: Add some form of debug option
I have a program that exposes a DBus service. While implementing the service, I've messed up types and signatures a few times, and that was hard to debug.

To properly debug what was going on, I had to modify the dbus library to add debug prints, to see what calls were being made, which objects were being looked up, what responses were being sent out, et cetera.

It would be great if the library had some built-in support for debugging or extra logging like this, so I don't have to modify it each time.

The simplest form would be a `ConnOption` which turns on debugging, and then all those details get printed out via the `log` package.

A perhaps more versatile API would be a series of `ConnOption`s which set up callback functions for each kind of activity. For example:

```
conn, err := dbus.SessionBusPrivate(dbus.WithLookupInspector(func(v interface{}) {
    log.Println("lookup:", v)
}))
```

The name and `v interface{}` are just made up for the sake of the example.

### PR DESCRIPTION
**Pull request by amenzhinsky**
Closes #160

### CODE REVIEW

File: conn.go [hunk 1]
```diff
@@ -635,18 +658,6 @@ func getTransport(address string) (transport, error) {
 	return nil, err
 }
 
-// dereferenceAll returns a slice that, assuming that vs is a slice of pointers
```
Comment (jsouthworth) [inline conn.go]: This should really be removed in a separate commit as it is unrelated to adding the interceptors.

File: conn.go [hunk 2]
```diff
@@ -383,11 +407,10 @@ func (conn *Conn) Object(dest string, path ObjectPath) BusObject {
 	return &Object{conn, dest, path}
 }
 
-func (conn *Conn) sendMessage(msg *Message) {
```
Comment (jsouthworth) [inline conn.go]: Why is this refactor necessary? If we keep it, it should be done in a separate commit as it is unrelated to adding the interceptors.

Comment (amenzhinsky) [reply conn.go]: I wanted to have one method for all outgoing messages, done as a separate commit

File: conn.go (hunk 1 above)
Comment (amenzhinsky) [reply conn.go]: This function is not used and it's not exported, extracted the change to a separate commit

**Comment by jsouthworth** [comment]
Thanks!
